### PR TITLE
[#155512172] Users can be created from an API endpoint

### PR DESCRIPTION
**Pull request by iankigen**
#### What does this PR do?

Allow users to be created from an API endpoint

#### Description of Task to be completed?

- add test for functionality
- add user serializer
- add user viewset
- add user url patterns

#### How should this be manually tested?

`python manage.py test`
or 
endpoint `/api/v1/user` in your browser

#### Any background context you want to provide?

only admin user can add users
#### What are the relevant pivotal tracker stories?

[#155512172](https://www.pivotaltracker.com/n/projects/2146417/stories/155512172)

#### Screenshots 
user-home

single-user

### CODE REVIEW

**Comment by iankigen** [comment]
@egichuri @alexkiura @nelmandela @HawiCaesar @Awinja-Andela @jmnyarega Kindly review

File: art/urls.py [hunk 1]
```diff
@@ -18,9 +18,12 @@
 from django.conf.urls import include
 from django.urls import path
 
+from api import urls
+
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    path('', include(urls)),
```
Comment (egichuri) [inline art/urls.py]: let's handle api versioning (api/v1) here for consistency.

File: api/serializers.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import serializers
+
+from .models import User
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = "__all__"
```
[the remark this answers is not in the document]
Comment (iankigen) [reply api/serializers.py]: alright. I'm working on it

File: api/tests/test_user.py [hunk 3]
```diff
@@ -1,7 +1,9 @@
 from django.test import TestCase
 from django.contrib.auth import get_user_model
+from rest_framework.test import APIClient
 
 User = get_user_model()
+client = APIClient()
 
 
 class UserTestCase(TestCase):
```
Comment (alexkiura) [inline api/tests/test_user.py]: Let's also add tests to ensure that only `POST` & `GET` are allowed for this endpoint.

Comment (iankigen) [reply api/tests/test_user.py]: on it

**Comment by iankigen** [comment]
@egichuri @alexkiura @nelmandela @HawiCaesar @Awinja-Andela Kindly review

**Comment by nelmandela** [comment]
LGTM

File: api/urls.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from rest_framework.routers import SimpleRouter
+
+from .views import UserViewSet
+
+router = SimpleRouter()
+router.register('user', UserViewSet)
```
Comment (egichuri) [inline api/urls.py]: let's go with `users` here

Comment (iankigen) [reply api/urls.py]: working on it

**Comment by iankigen** [comment]
@alexkiura Kindly review

File: api/tests/test_user.py (hunk 3 above)
Comment (alexkiura) [reply api/tests/test_user.py]: Looks good.

**Pull request merged**